src/app/clusters/level-control/level-control.cpp

```diff
@@ -123,6 +123,145 @@ static bool shouldExecuteIfOff(EndpointId endpoint, CommandId commandId,
                                chip::Optional<chip::BitMask<LevelControlOptions>> optionsMask,
                                chip::Optional<chip::BitMask<LevelControlOptions>> optionsOverride);
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+class DefaultLevelControlSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    enum LevelControlEFS : uint8_t
+    {
+        kLevel = 0,
+        kFrequency,
+    };
+
+    // As per spec, 2 attributes are scenable in the level control cluster
+    static constexpr uint8_t scenableAttributeCount = 2;
+
+    DefaultLevelControlSceneHandler() = default;
+    ~DefaultLevelControlSceneHandler() override {}
+
+    // Default function for LevelControl cluster, only puts the LevelControl cluster ID in the span if supported on the caller
+    // endpoint
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        ClusterId * buffer = clusterBuffer.data();
+        if (emberAfContainsServer(endpoint, LevelControl::Id) && clusterBuffer.size() >= 1)
+        {
+            buffer[0] = LevelControl::Id;
+            clusterBuffer.reduce_size(1);
+        }
+        else
+        {
+            clusterBuffer.reduce_size(0);
+        }
+    }
+
+    // Default function for LevelControl cluster, only checks if LevelControl is enabled on the endpoint
+    bool SupportsCluster(EndpointId endpoint, ClusterId cluster) override
+    {
+        return (cluster == LevelControl::Id) && (emberAfContainsServer(endpoint, LevelControl::Id));
+    }
+
+    /// @brief Serialize the Cluster's EFS value
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serializedBytes data to serialize into EFS
+    /// @return CHIP_NO_ERROR if successfully serialized the data, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR SerializeSave(EndpointId endpoint, ClusterId cluster, MutableByteSpan & serializedBytes) override
+    {
+        using AttributeValuePair = Scenes::Structs::AttributeValuePair::Type;
+
+        app::DataModel::Nullable<uint8_t> level;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::CurrentLevel::Get(endpoint, level), CHIP_ERROR_READ_FAILED);
+        uint16_t frequency;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::CurrentFrequency::Get(endpoint, &frequency),
+                            CHIP_ERROR_READ_FAILED);
+
+        AttributeValuePair pairs[scenableAttributeCount];
+
+        // Check for nullable
+        if (!level.IsNull())
```

Comment:
is there an ordering requirement? Otherwise you could always encode frequency in pairs[0] and only conditionally encode level at position 1.

Reply:
Right, we don't have an order requirement since we are using the explicit form. However now that the null encoding has been define I suppose we do want to know if the current level is null so I'll encode it to maxLevel + 1 instead.